src/app/clusters/unit-localization-server/unit-localization-server.h

```diff
@@ -0,0 +1,62 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app-common/zap-generated/cluster-enums.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/data-model/List.h>
+#include <cstdint>
+#include <lib/core/CHIPError.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace UnitLocalization {
+
+inline constexpr uint16_t kClusterRevision              = 2;
+inline constexpr uint8_t kMinSupportedLocalizationUnits = 2;
+inline constexpr uint8_t kMaxSupportedLocalizationUnits = 3;
+
+class UnitLocalizationServer : public AttributeAccessInterface
+{
+public:
+    // Register for the UnitLocalization cluster on all endpoints.
+    static UnitLocalizationServer & Instance();
+
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+
+    CHIP_ERROR SetSupportedTemperatureUnits(DataModel::List<TempUnitEnum> & units);
+    DataModel::List<TempUnitEnum> GetSupportedTemperatureUnits(void) { return mSupportedTemperatureUnits; }
+
+private:
+    UnitLocalizationServer() : AttributeAccessInterface(Optional<EndpointId>::Missing(), UnitLocalization::Id) {}
+
+    CHIP_ERROR ReadSupportedTemperatureUnits(EndpointId endpoint, AttributeValueEncoder & aEncoder);
+    CHIP_ERROR ReadClusterRevision(EndpointId endpoint, AttributeValueEncoder & aEncoder);
+
+    static UnitLocalizationServer mInstance;
+    DataModel::List<TempUnitEnum> mSupportedTemperatureUnits{ DataModel::List<TempUnitEnum>(mUnitsBuffer) };
```

Comment:
Maybe storing here the `size_t` of actually used `mUnitsBuffer` entries instead of `DataModel::List<TempUnitEnum>` will be easier to read?

Reply:
I agree that this is a simple task here but considering the attribute itself is a list - the readability is good, I would say. This also makes it easier to pass around in function calls.